examples/tests/NavigationSampleTests.elm

```diff
@@ -0,0 +1,35 @@
+module NavigationSampleTests exposing (all)
+
+import Expect
+import NavigationSample exposing (program)
+import Test exposing (..)
+import Test.Html.Event as Event
+import Test.Html.Query exposing (..)
+import Test.Html.Selector exposing (..)
+import TestContext exposing (..)
+
+
+all : Test
+all =
+    describe "NavigationSample"
+        [ test "renders navigation history" <|
+            \() ->
+                NavigationSample.program
+                    |> start
+                    |> simulate (find [ class "bears" ]) Event.click
+                    |> simulate (find [ class "cats" ]) Event.click
+                    |> expectView
+                    |> find [ class "history" ]
+                    |> Expect.all
+                        [ has [ text "#bears" ]
+                        , has [ text "#cats" ]
+                        ]
+        , test "works for user initiated navigation" <|
+            \() ->
+                NavigationSample.program
+                    |> start
+                    |> navigate "#cats"
```

Comment:
Same question as above; I don't think we can limit the API to only working with the hash part of URLs.  Does the API work with full URLs?  If so, I think the example should show what that would look like.  If not, we should think about how to deal with that.